cpp/simulations/abm_braunschweig.cpp

```diff
@@ -0,0 +1,267 @@
+        break;
+    case 5:
+        type = mio::abm::LocationType::Home; // Private Erledigung
+        break;
+    case 6:
+        type = mio::abm::LocationType::SocialEvent; // Sonstiges
+        break;
+    case 7:
+        type = mio::abm::LocationType::Home;
+        break;
+    }
+    return type;
+}
+
+void create_world_from_data(mio::abm::World& world, const std::string& filename)
+{
+    // File pointer
+    std::fstream fin;
+
+    // Open an existing file
+    fin.open(filename, std::ios::in);
+    std::vector<int32_t> row;
+    std::vector<std::string> row_string;
+    std::string line;
+
+    getline(fin, line);
+    std::vector<std::string> titles;
+    boost::split(titles, line, boost::is_any_of(";"));
+    uint32_t count                        = 0;
+    std::map<std::string, uint32_t> index = {};
+    for (auto const& title : titles) {
+        index.insert({title, count});
+        row_string.push_back(title);
+        count++;
+    }
+    std::map<uint32_t, mio::abm::LocationId> locations = {};
+    std::map<uint32_t, mio::abm::Person&> persons      = {};
+
+    while (getline(fin, line)) {
+        row.clear();
+
+        // read columns in this row
+        split_line(line, &row);
+
+        uint32_t person_id   = row[index["puid"]];
+        uint32_t age         = row[index["age"]]; // TODO
+        uint32_t location_id = row[index["loc_id_end"]];
+        uint32_t activity    = row[index["activity_end"]];
+
+        auto it_location              = locations.find(location_id);
+        mio::abm::LocationId location = it_location->second;
+        if (it_location == locations.end()) {
+            location = world.add_location(get_location_type(activity), 0); // TODO: adjust LocationType
+            locations.insert({location_id, location});
+        }
+        auto it_person = persons.find(person_id);
+        if (it_person == persons.end()) {
+            auto& person =
+                world.add_person(location, mio::abm::InfectionState::Susceptible, static_cast<mio::abm::AgeGroup>(age));
+            persons.insert({person_id, person});
+            it_person = persons.find(person_id);
+        }
+        it_person->second.set_assigned_location(location);
+    }
+}
+
+/**
+ * Create a sampled simulation with start time t0.
+ * @param t0 the start time of the simulation
+*/
+mio::abm::Simulation create_sampled_simulation(const mio::abm::TimePoint& t0)
+{
+
+    // Assumed percentage of infection state at the beginning of the simulation.
+    // ScalarType exposed_pct = 0.005, infected_pct = 0.001, carrier_pct = 0.001, recovered_pct = 0.0;
+
+    //Set global infection parameters (similar to infection parameters in SECIR model) and initialize the world
+    mio::abm::GlobalInfectionParameters infection_params;
+    auto world = mio::abm::World(infection_params);
+
+    // Create the world object from statistical data.
+    create_world_from_data(world, "../../../Documents/HZI/memilio/data/mobility/bs.csv");
```

Comment:
This leaks some data about your internal file history. Consider deleting this soon.

Reply:
Replaced by relative path.